pkg/networkserver/downlink_test.go

```diff
@@ -852,16 +852,26 @@ func TestProcessDownlinkTask(t *testing.T) {
 					}(),
 					func(paths ...*ttnpb.DownlinkPath) *ttnpb.TxRequest {
 						return &ttnpb.TxRequest{
-							Class:             ttnpb.CLASS_A,
-							DownlinkPaths:     paths,
-							Priority:          ttnpb.TxSchedulePriority_HIGHEST,
-							Rx1DataRateIndex:  lastUp.Settings.DataRateIndex,
-							Rx1Delay:          dev.MacState.CurrentParameters.Rx1Delay,
-							Rx1Frequency:      dev.MacState.CurrentParameters.Channels[lastUp.DeviceChannelIndex].DownlinkFrequency,
-							Rx2DataRateIndex:  dev.MacState.CurrentParameters.Rx2DataRateIndex,
+							Class:         ttnpb.CLASS_A,
+							DownlinkPaths: paths,
+							Priority:      ttnpb.TxSchedulePriority_HIGHEST,
+							Rx1DataRate:   &lastUp.Settings.DataRate,
+							Rx1Delay:      dev.MacState.CurrentParameters.Rx1Delay,
+							Rx1Frequency:  dev.MacState.CurrentParameters.Channels[lastUp.DeviceChannelIndex].DownlinkFrequency,
+							Rx2DataRate: &ttnpb.DataRate{
+								Modulation: &ttnpb.DataRate_Lora{
+									Lora: &ttnpb.LoRaDataRate{
+										Bandwidth:       125000,
+										SpreadingFactor: 12 - uint32(dev.MacState.CurrentParameters.Rx2DataRateIndex),
```

Comment:
This isn't really beautiful but I think this is acceptable in testing. Otherwise we'd need a utility function for this. Note that in not all cases we have PHY available and we only do this for RX2 which is always LoRa (not FSK and not LR-FHSS)

Reply:
It's acceptable but at the same time we will look back at this in the future and feel repulsion. 

Consider https://github.com/TheThingsNetwork/lorawan-stack/commit/78844dbc531d6d9b4853b4bf64bfe8b94a08b093 - maybe you like this more.